kmir/src/kmir/__main__.py

```diff
@@ -69,23 +90,39 @@ def _kmir_gen_spec(opts: GenSpecOpts) -> None:
         sys.exit(1)
 
     kmir_kast, _ = parse_result
-    config = tools.make_init_config(kmir_kast, opts.start_symbol)
+    config = tools.make_init_config(kmir_kast, opts.start_symbol, 'KmirCell')
     config_with_cell_vars, _ = split_config_from(config)
 
     lhs = CTerm(config)
-    new_k_cell = tools.kmir.Symbols.END_PROGRAM
+    new_k_cell = KMIR.Symbols.END_PROGRAM
     rhs = CTerm(Subst({'K_CELL': new_k_cell})(config_with_cell_vars))
-    claim, _ = cterm_build_claim(opts.input_file.stem, lhs, rhs)
+    claim, _ = cterm_build_claim(opts.input_file.stem.replace('_', '-'), lhs, rhs)
 
     output_file = opts.output_file
     if output_file is None:
-        output_file = opts.input_file.with_suffix('.spec.json')
+        suffixes = ''.join(opts.input_file.suffixes)
```

Comment:
Small Q: Maybe I misunderstand the python code here, but should this maybe rather be `'.'.join`?

Reply:
The elements in `suffixes` contain the dots.